android/src/main/java/com/inbrain/InBrainSurveysModule.java

```diff
@@ -265,13 +266,24 @@ public void nativeSurveysReceived(List<Survey> surveys) {
                         map.putBoolean("currencySale", survey.currencySale);
                         map.putDouble("multiplier", survey.multiplier);
 
-
                         WritableArray categories = Arguments.createArray();
+                        WritableArray namedCategories = Arguments.createArray();
+                        WritableMap categoryNamed = Arguments.createMap();
```

Comment:
If I understood correctly - the "id" and "name" values will be re-written at each "for" interaction. And that variable is used only inside "for" loop - so probably we can move it's declaration inside the loop? Just to keep the local variable only where it needed